demisto_sdk/commands/common/hook_validations/base_validator.py

```diff
@@ -61,3 +66,37 @@ def handle_error(self, error_massage, error_code, file_path, should_print=True,
                 click.secho(formatted_error, fg="bright_red")
 
         return formatted_error
+
+    def check_deprecated(self, file_path):
+        if file_path.endswith('.yml'):
+            yml_dict = get_yaml(file_path)
+            if ('deprecated' in yml_dict and yml_dict['deprecated'] is True) or \
+                    (os.path.basename(file_path).startswith('playbook') and 'hidden' in yml_dict and
+                     yml_dict['hidden'] is True):
+                self.add_flag_to_ignore_list(file_path, 'deprecated')
```

Comment:
no need to read the file all these times, let's store this info somewhere and make sure we are reading it only once.